kit/platform/id.go

```diff
@@ -143,3 +144,20 @@ func (i ID) MarshalText() ([]byte, error) {
 func (i *ID) UnmarshalText(b []byte) error {
 	return i.Decode(b)
 }
+
+// Value implements the database/sql Valuer interface for adding IDs to a sql database.
+func (i ID) Value() (driver.Value, error) {
+	return i.String(), nil
+}
+
+// Scan implements the database/sql Scanner interface for retrieving IDs from a sql database.
+func (i *ID) Scan(value interface{}) error {
+	switch v := value.(type) {
+	case int64:
+		return i.DecodeFromString(strconv.Itoa(int(v)))
+	case string:
+		return i.DecodeFromString(v)
+	default:
+		return ErrInvalidID
+	}
+}
```

Comment:
Adding the `Value` and `Scan` methods makes it so `sqlx` can read directly from the values into the database and back out of the database into the Go values when doing queries. Otherwise you need a lot of conversion code in each query.

Reply:
No floating-point conversions?

Reply:
Yeah this is really an artifact of how the values are stored and come out of the database. The IDs are stored in the DB as their hex encoded string, like "1234affa12341234". A value like that comes out and gets interpreted by the driver as a string, so we can scan it back into a `platform.ID` structure using the `DecodeFromString` method. I found in my testing that an ID like "1234123412341234" gets interpreted by the driver as an `int64` during the query and scan, which is why there is a case for that. It's weird to me that even though the DB column type is string it was still coming back as an `int64`, but so it goes. There shouldn't be any other types like floats to deal with since the hex encoded string of an ID shouldn't ever have a decimal.